code/game/machinery/computer/arcade.dm

```diff
@@ -189,107 +250,301 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
-		player_hp = 30
-		player_mp = 10
-		enemy_hp = 45
-		enemy_mp = 20
-		gameover = FALSE
-		turtle = 0
+		temp = "<br><center><h3>New Round<center><h3>"
 
 		if(obj_flags & EMAGGED)
 			Reset()
 			obj_flags &= ~EMAGGED
 
+		enemy_setup(gamerSkill)
+		screen_setup(usr)
+
+
 	add_fingerprint(usr)
 	updateUsrDialog()
 	return
 
-/obj/machinery/computer/arcade/battle/proc/arcade_action(mob/user)
+
+///happens after a player action and before the enemy turn. the enemy turn will be cancelled if there's a gameover.
+/obj/machinery/computer/arcade/battle/proc/arcade_action(mob/user,player_stance,attackamt)
+	screen_setup(user)
+	blocked = TRUE
+	if(player_stance == "attack" || player_stance == "power_attack")
+		if(attackamt > 40)
+			playsound(src, 'sound/arcade/boom.ogg', 50, TRUE, extrarange = -3)
+		else
+			playsound(src, 'sound/arcade/hit.ogg', 50, TRUE, extrarange = -3)
+
+	timer_id = addtimer(CALLBACK(src, .proc/enemy_action,player_stance,user),1 SECONDS,TIMER_STOPPABLE)
+	gameover_check(user)
+
+
+///the enemy turn, the enemy's action entirely depend on their current passive and a teensy tiny bit of randomness
+/obj/machinery/computer/arcade/battle/proc/enemy_action(player_stance,mob/user)
+	temp = ""
+
+	if(LAZYLEN(last_three_move) < 3) //we keep the last three action of the player in a list here
+		LAZYADD(last_three_move, player_stance)
+
+	else if(LAZYLEN(last_three_move) == 3)
+		LAZYREMOVE(last_three_move, last_three_move[1])
+		LAZYADD(last_three_move, player_stance)
+
+	else if(LAZYLEN(last_three_move) > 3)
+		last_three_move = null //this shouldn't even happen but we empty the list if it somehow goes above 3
+
+	var/enemy_stance
+	var/attack_amount = rand(8,10) //making the attack amount not vary too much so that it's easier to see if the enemy has a shotgun
+
+	if(player_stance == "defend")
+		attack_amount -= 5
+
+	//if emagged, cuban pete will set up a bomb acting up as a timer. when it reaches 0 the player fucking dies
+	if(obj_flags & EMAGGED)
+		if(bomb_cooldown == 18)
+			temp += "<br><center><h3>[enemy_name] takes two valve tank and links them together, what's he planning?<center><h3>"
+		if(bomb_cooldown == 15)
+			temp += "<br><center><h3>[enemy_name] adds a remote control to the tan- ho god is that a bomb?<center><h3>"
+		if(bomb_cooldown == 12)
+			temp += "<br><center><h3>[enemy_name] throws the bomb next to you, you'r too scared to pick it up. <center><h3>"
+		if(bomb_cooldown == 6)
+			temp += "<br><center><h3>[enemy_name]'s hand brushes the remote linked to the bomb, your heart skipped a beat. <center><h3>"
+		if(bomb_cooldown == 2)
+			temp += "<br><center><h3>[enemy_name] is going to press the button! It's now or never! <center><h3>"
+		if(bomb_cooldown == 0)
+			player_hp -= attack_amount * 1000 //hey it's a maxcap we might as well go all in
+		bomb_cooldown--
+
+	//yeah I used the shotgun as a passive, you know why? because the shotgun gives +5 attack which is pretty good
+	if("shotgun" in current_enemy_passive)
```

Comment:
Try using associative lists, so you can check with `current_enemy_passive["shotgun"]` or with `LAZYACCESS(current_enemy_passive, "shotgun")`